docs/develop/python/message-passing.mdx

```diff
@@ -369,13 +423,42 @@ For example, in an `async def` handler it could be used to wait until the handle
         )
 ```
 
-This is necessary if your Signal or Update handlers require something in the main Workflow method to be done first, since a Signal or Update handler can execute before the main Workflow method has started.
+Remember: Handlers can execute before the main Workflow method starts.
+
+### Ensure your handlers finish before the Workflow completes {#wait-for-message-handlers}
+
+Workflow wait conditions let your handlers complete before a Workflow finishes.
```

Comment:
```suggestion
Workflow wait conditions can be used to ensure your handlers complete before the Workflow finishes.
```

However, I'm not sure it's needed as the others sentences do a good job.

Reply:
Summary: The first sentence is clearer and more readable due to its brevity and directness. The second sentence, while still clear, introduces additional complexity that may slightly impact readability.

Stet

Reply:
"let" seems wrong to me; I believe it is synonymous with "allow" and thus carries with it the possibility that the allowed action will not be taken.

 But here using the wait condition guarantees that the handler _will_ complete before the workflow completes; there is no possibility that it will opt not to do that.

Reply:
I'll give you "can ensure"

Reply:
Sounds good!